server/platform/services/sharedchannel/channelinvite.go

```diff
@@ -255,7 +255,25 @@ func (scs *Service) createDirectChannel(invite channelInviteMsg) (*model.Channel
 		return nil, fmt.Errorf("cannot create direct channel `%s` insufficient participant count `%d`", invite.ChannelId, len(invite.DirectParticipantIDs))
 	}
 
-	channel, err := scs.app.GetOrCreateDirectChannel(request.EmptyContext(scs.server.Log()), invite.DirectParticipantIDs[0], invite.DirectParticipantIDs[1], model.WithID(invite.ChannelId))
+	user1 := invite.DirectParticipantIDs[0]
+	user2 := invite.DirectParticipantIDs[1]
+
+	// ensure users can see each other (allowed to DM each other)
+	canSee, appErr := scs.app.UserCanSeeOtherUser(request.EmptyContext(scs.server.Log()), user1, user2)
```

Comment:
Shouldn't we check this both ways -- that `user1` can see `user2`, but also that `user2` can see `user1`? Or maybe check that one of the users is remote and that the remote user can see the other user, because the remote is what initiated the channel creation here?

Reply:
@jupenur Agreed.  I made the check stronger and defined the rules in comments.